contributions.md

```diff
@@ -53,10 +53,36 @@ If you discover a security vulnerability within Laravel, please send an e-mail t
 
 Laravel follows the [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md) coding standard and the [PSR-4](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-4-autoloader.md) autoloading standard.
 
+### Additional requirements
+
+Short definition syntax for arrays is prefered.
+
+    $array = ['one', 'two', 'three']; //Good
+    $array = array('one', 'two', 'three'); //Bad
+
+When using class names, `::class` constant should be used in most cases.
+
+    $className = Request::class; //Good
+    $className = \Illuminate\Http\Request::class; //Good too
+    $className = 'Illuminate\Http\Request'; //Bad
+
+All imported classes and namespaces should be ordered by length.
+
+    namespace Illuminate\Foo;
+    
+    use Illuminate\Support\Str;
+    use Illuminate\Support\Manager;
+    use JohnDoe\ExternalLibrary\OneClass;
+    use JohnDoe\ExternalLibrary\TwoClass;
+    use Illuminate\Database\Eloquent\Model;
+    use JohnDoe\ExternalLibrary\Model as JohnDoeModel;
+
 ### DocBlocks
 
 `@param` tags should **not be aligned** and arguments should be separated by **2 spaces**. 
 
+Class hints should use fully qualified names with leading `\`.
```

Comment:
"and" is better rather than "with", since the `\` is not part of the fully qualified name